middle_end/flambda2/types/grammar/more_type_creators.ml

```diff
@@ -33,6 +33,15 @@ let unknown (kind : K.t) =
 
 let unknown_like t = unknown (TG.kind t)
 
+let unknown_from_shape (shape : K.Block_shape.t) index =
+  let kind =
+    match shape with
+    | Value_only -> K.value
+    | Float_record -> K.naked_float
+    | Mixed_record fields -> (K.Mixed_block_shape.field_kinds fields).(index)
+  in
+  unknown kind
```

Comment:
This should not be useful: should be replace at use site by `unknown (K.Block_shape.element_kind)`